### ISSUE TITLE
[Stats Refresh] Voiceover audit

### ISSUE THREAD

**Issue by frosty** [wordpress-mobile/WordPress-iOS#11995]
It's important that our new stats refresh is accessible for users who rely on VoiceOver. At the moment, it doesn't function as well as the previous version of stats. Let's do an audit, and note here any issues that need resolving. Note that these are not necessarily listed in priority order.

- [x] View More rows in stats tables are not read correctly
- [x] Reflect expanded / collapsed state of expandable rows
- [x] "No data for this period" rows are not read correctly
- [ ] Expanded rows are not accessible

### Date selector bar

- [x] Perhaps prefix the current period with text hinting that it is the currently displayed period. e.g. "Current period: 25 June 2019". Or as a hint: "25 June 2019. The current period displayed."
- [x] Date bar arrows just say 'button', but should describe their function.
- [x] The grey spacer row below date bar is highlightable / selectable. Would be good if we could skip it.

### Period Chart

- [ ] Difference label should hint that it's compared to the previous period.

### Insights

- [x] Heading for "Latest post summary" reads as "Latest P.O.S.T. summary". We may need to lowercase the accessibility description (#12201)
- [ ] Ordering of items in the latest post summary block is incorrect. It skips from the summary to View More, and then back to the items in the middle. "View More" is also read when VoiceOver summarises the entire block, in between the view count and the word "Views". (@frosty)
- [x] The Today block and the All Time Block have an invisible button at the bottom that gets highlighted – we should skip this.
- [ ] Annual site stats block isn't read correctly.
- [x] Most popular time block could improve the description of the most popular time – for me it reads "14, 5% of views". "2pm" would be better than 14 when reading it. (#12201)
- [x] #13123 Posting activity could provide a better description, even if it's just "graphic depicting days when posts were published over the last 3 months". When navigating through the graphic itself, we should improve the description of each day if possible (currently just says 'dimmed button'). Ideally: skip empty days, and note the number of posts on the days with posts. I don't know if we have this info available or not. 
- [ ] Posting Activity: Improve #13123. Possibly add a visual indicator of what date is currently being spoken. The red selection is not happening when you navigate using VoiceOver.
- [ ] Posting Activity: From #13123. The container's accessibilityLabel could describe what you can visually see. For example, “December 2019. High posting activity“. Or perhaps the total number of posts for the month?
- [ ] Reading order for Comments table is slightly off – table and rows are read before the tabs at the top. Same for Followers table (same style).
- [ ] Add Stats Card's close button has no label
- [ ] Tags and Categories expanded items are not accessible

**Comment by shiki** [comment]
Hi @frosty, @danielebogo (and probably @ScoutHarris) I'm gonna take a look at improving VoiceOver usage for the Posting Activity. 

Just letting you know in case of conflicts (you're assigned to this issue). 🙂

**Comment by pmciano** [comment]
We have a user that had some feedback about VoiceOver and the iOS WordPress app in 2764271-zen and they included a screen recording of their experience. I'm posting here in case this helps in identifying areas for improvement in our implementation.

**Comment by staskus** [comment]
I'm closing this outdated task related to an old project.

Although we need to continuously improve VoiceOver and some of the feedback here may still be valid, there were improvements made while working on Stats Revamp project to address some of the main issues:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/19773
- https://github.com/wordpress-mobile/WordPress-iOS/pull/19368

The future VoiceOver tasks should be created for a new project or from a new piece of user feedback.

cc @guarani
